website.js

```diff
@@ -18,6 +18,7 @@ var clientID        = options.identity.clientId
 var secret          = options.identity.clientSecret
 var redirect        = options.identity.redirectUrl
 var io              = require('socket.io')(server);
+var ip;
```

Comment:
Be careful when re-using the `ip` in this case, because it's unreliable.

If there are two users (`x` and `y`) and they both enter the website. You're caching the ip here and re-using it in your sockets.

If user `x` does a request, it sets the ip to his ip (ex `1.2.3.4`),
User `x` does a `addResult` socket request, but doesn't finish yet. 
In the meantime... user `y` does a request, it sets the ip to his ip now (ex `4.3.2.1`)
User `x`'s socket request finishes now, but the ip is now set to `4.3.2.1`, which is not user `x`'s IP address.

Maybe I am missing something here, but this might be a very dangerous thing to do 😉